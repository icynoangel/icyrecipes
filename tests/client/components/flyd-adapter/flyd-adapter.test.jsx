import React from 'react';
import ReactDOM from 'react-dom';
import {mount} from 'enzyme';
import FlydAdapter, {
  stream
} from './../../../../src/client/js/components/flyd-adapter/flyd-adapter';
import enzymeToJson from 'enzyme-to-json';

describe('<FlydAdapter />', function() {
  this.getComponent = type => {
    return <FlydAdapter stream={this.properties} component={this.component} />;
  };

  beforeEach(() => {
    this.properties = stream({
      average: 5,
      colors: ['blue', 'yellow']
    });
    this.component = function(props) {
      return (
        <div>
          <div>Average: {props.average}</div>
          <div>
            {props.colors.map((color, index) => {
              return <div key={index}>Color: {color}</div>;
            })}
          </div>
        </div>
      );
    };
  });

  it('Should render wrapped component', () => {
    const wrapper = mount(this.getComponent());
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });

  it('Should render wrapped component and update its props', () => {
    const wrapper = mount(this.getComponent());
    this.properties({
      average: 10,
      colors: ['red', 'green']
    });
    wrapper.update();
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });
});
