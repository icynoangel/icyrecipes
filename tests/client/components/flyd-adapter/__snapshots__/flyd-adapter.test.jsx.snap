// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlydAdapter /> Should render wrapped component 1`] = `
<FlydAdapter
  component={[Function]}
  stream={[Function]}
>
  <Component
    average={5}
    colors={
      Array [
        "blue",
        "yellow",
      ]
    }
  >
    <div>
      <div>
        Average: 
        5
      </div>
      <div>
        <div
          key="0"
        >
          Color: 
          blue
        </div>
        <div
          key="1"
        >
          Color: 
          yellow
        </div>
      </div>
    </div>
  </Component>
</FlydAdapter>
`;

exports[`<FlydAdapter /> Should render wrapped component and update its props 1`] = `
<FlydAdapter
  component={[Function]}
  stream={[Function]}
>
  <Component
    average={10}
    colors={
      Array [
        "red",
        "green",
      ]
    }
  >
    <div>
      <div>
        Average: 
        10
      </div>
      <div>
        <div
          key="0"
        >
          Color: 
          red
        </div>
        <div
          key="1"
        >
          Color: 
          green
        </div>
      </div>
    </div>
  </Component>
</FlydAdapter>
`;
